### PR TITLE
Replaced distutils module to sysconfig

### PR DESCRIPTION
**Pull request by asenyaev**
This PR fixes following issue https://github.com/opencv/opencv/issues/21141.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: cmake/OpenCVDetectPython.cmake [hunk 1]
```diff
@@ -177,7 +177,7 @@ if(NOT ${found})
 
     if(NOT ANDROID AND NOT IOS)
       if(CMAKE_HOST_UNIX)
-        execute_process(COMMAND ${_executable} -c "from distutils.sysconfig import *; print(get_python_lib())"
+        execute_process(COMMAND ${_executable} -c "from sysconfig import *; print(get_python_lib())"
```
Comment (alalek) [inline cmake/OpenCVDetectPython.cmake:180]: Looks like this doesn't work as expected (Linux x64):
```
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.6", minimum required is "2.7") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found suitable exact version "2.7.6") 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'get_python_lib' is not defined
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.4.3", minimum required is "3.2") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.4m.so (found suitable exact version "3.4.3") 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'get_python_lib' is not defined
```

Comment (asenyaev) [reply cmake/OpenCVDetectPython.cmake]: I'm sorry, I didn't change the function here. Fixed it.

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Comment by alalek** [comment]
BTW, Looks like for Python <=3.9 patch changes binary names:
- [before](http://pullrequest.opencv.org/buildbot/builders/3_4_winpack-build-win64-vc14/builds/1292): `python-3.5/cv2.cp35-win_amd64.pyd`
- [after](http://pullrequest.opencv.org/buildbot/builders/3_4_winpack-build-win64-vc14/builds/1293): `python-3.5/cv2.pyd`

Name for Python 3.10 is wider: `python-3.10/cv2.cp310-win_amd64.pyd`

**Pull request merged**